ImageLounge/src/DkCore/DkUtils.cpp

```diff
@@ -981,11 +981,112 @@ QString DkUtils::stdWStringToQString(const std::wstring &str)
 }
 
 // DkConvertFileName --------------------------------------------------------------------
-DkFileNameConverter::DkFileNameConverter(const QString &fileName, const QString &pattern, int cIdx)
+DkFileNameConverter::DkFileNameConverter(const QString &p)
+    : mFrags{}
 {
-    this->mFileName = fileName;
-    this->mPattern = pattern;
-    this->mCIdx = cIdx;
+    const QStringView pattern(p);
+    qsizetype start = 0;
+    Token type = Token::Text;
+    std::vector<uint> numbers{};
+    FragType fragType = FragType::Text;
+
+    for (qsizetype i = 0; i < pattern.size(); i++) {
+        const QChar ch = pattern.at(i);
+
+        if (ch != '<' && ch != '>' && ch != ':') {
+            continue;
+        }
+
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)

```

Comment:
I think we should aim to avoid conditional code as much as possible unless there is a meaningful performance or functional impact. 

I believe QString:mid() works on either version here and also for left()/first()

Reply:
These are `QStringView`s, and the `mid` method is [deprecated in Qt6](https://doc.qt.io/qt-6/qstringview-obsolete.html#mid).

For `QString::mid()`, there are performance gains according to [the documentation](https://doc.qt.io/qt-6/qstring.html#mid) if the indices are within bounds. 

Because the API is different enough, I think we should drop support of Qt5 once things stabilize. Although these conditionals are ugly, they are easy to spot and can make removing Qt5 code as easy as removing one branch, and we are not losing things due to being locked to old APIs.